src/DynamoCoreWpf/Manipulation/Manipulator.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Autodesk.DesignScript.Interfaces;
+using Dynamo.Wpf.Interfaces;
+
+namespace Dynamo.Wpf.Manipulation
+{
+    public abstract class NodeManipulatorCreator : INodeManipulatorCreator
+    {
+        /// <summary>
+        /// Create a set of manipulators for the given node by inspecting its numeric inputs
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="manipulatorContext"></param>
+        /// <returns></returns>
+        public IManipulator Create(Models.NodeModel node, DynamoManipulatorContext manipulatorContext)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public abstract class Manipulator : IManipulator
+    {
+        public abstract void Dispose();
+
+        public abstract void Tessellate(IRenderPackage package, TessellationParameters parameters);
+    }
+
+    internal class CompositeManipulator : Manipulator
+    {
+        private readonly List<IManipulator> subManipulators;
+        public CompositeManipulator(List<IManipulator> manipulators)
```

Comment:
This could be generalized to `IEnumerable`. And then copied in the constructor.  That way, client code does not have a mutable copy of `subManipulators`.